lib/utils/assertType.js

```diff
@@ -23,6 +23,50 @@ function assertObject(attr, data) {
   return data;
 }
 
+/**
+ * Throws if the provided object is not an object or if it contains heterogeaous typed properties.
+ * Returns the unmodified data if validated.
+ *
+ * @throws {ParseError}
+ * @param {string} attr - tested attribute name
+ * @param {*} data
+ * @param {string} [type] - expected type for data properties
+ * @returns {object|null}
+ */
+function assertUniTypeObject(attr, data, type) {
+  data = assertObject(attr, data);
+
+  if (data === null) {
+    return null;
+  }
+
+  Object.keys(data).forEach(key => {
+    if (type === undefined) {
+      type = typeof data[key];
+    }
+    type = type.toLowerCase();
+
+    let msg = `Attribute ${attr} must be of type "object" and have all its properties of type ${type}.`
+      + `\nExpected "${attr}.${key}" to be of type "${type}", but go "${typeof data[key]}".`;
+
+    switch(type) {
```

Comment:
Actually, I had not thought about it. I just needed to test an object containing strings.
Strictly speaking, null is an object and I am not against the idea of considering it a valid value.

Further, as this method does not rely on `intanceof`, I don't see any use case where one would need to check an object has all its properties of type `object`.